superset/views/core.py

```diff
@@ -1645,19 +1645,16 @@ def dashboard(  # pylint: disable=too-many-locals
             request.args.get(utils.ReservedUrlParameters.EDIT_MODE.value) == "true"
         )
 
-        # Hack to log the dashboard_id properly, even when getting a slug
-        @event_logger.log_this
-        def dashboard(**_: Any) -> None:
-            pass
-
-        dashboard(
+        event_logger.log_with_context(
```

Comment:
i found this only logs a very short time, can we log the duration of the whole `dashboard` method?

Reply:
Good catch! Did some refactor and added a couple of unit tests to make sure this doesn't slip through the cracks next time.